src/tool/fy-tool.c

```diff
@@ -2392,7 +2405,71 @@ int main(int argc, char *argv[])
 			} else {
 				while ((fyev = fy_parser_parse(fyp)) != NULL) {
 					if (!null_output) {
-						rc = fy_emit_event_from_parser(fye, fyp, fyev);
+						if (recreating) {
+							fyeev = NULL;
+							switch (fyev->type) {
+								case FYET_STREAM_START:
+								case FYET_STREAM_END:
+								case FYET_MAPPING_END:
+								case FYET_SEQUENCE_END:
+									fyeev = fy_emit_event_create(fye, fyev->type);
+									break;
+								case FYET_DOCUMENT_START:
+									fyeev = fy_emit_event_create(fye, FYET_DOCUMENT_START,
+								 				fyev->document_start.implicit,
+												fy_document_state_version_explicit(fyev->document_start.document_state)
+													? fy_document_state_version(fyev->document_start.document_state)
+													: NULL,
+												fy_document_state_tags_explicit(fyev->document_start.document_state)
+													? fy_document_state_tag_directives(fyev->document_start.document_state)
+												 	: NULL);
+									break;
+								case FYET_DOCUMENT_END:
+									fyeev = fy_emit_event_create(fye, FYET_DOCUMENT_END,
+												fyev->document_end.implicit);
+									break;
+								case FYET_MAPPING_START:
+								case FYET_SEQUENCE_START:
+									fyeev = fy_emit_event_create(fye, fyev->type,
+												fy_event_get_node_style(fyev),
+												fy_event_get_anchor_token(fyev)
+													? fy_token_get_text0(fy_event_get_anchor_token(fyev))
+													: NULL,
+												fy_event_get_tag_token(fyev)
+													? fy_tag_token_short0(fy_event_get_tag_token(fyev))
+													: NULL);
+									break;
+								case FYET_SCALAR:
+									eevlen = 0;
+									fyeev = fy_emit_event_create(fye, FYET_SCALAR,
+								 				fy_scalar_token_get_style(fy_event_get_token(fyev)),
+												fy_token_get_text(fy_event_get_token(fyev), &eevlen),
+												eevlen,
```

Comment:
This doesn't work.

The order of evaluation for function arguments is undefined in the C standard, but specifically in most compilers it is done in reverse order.

You are expecting eevlen to be set before passing it to the function as argument. What happens is that eevlen is still 0 when calling the method and results in all scalars to have 0 length.

Did you try running the tests in --debug mode (set when configuring).